packages/@glimmer/runtime/lib/compiled/opcodes/component.ts

```diff
@@ -699,14 +766,32 @@ APPEND_OPCODES.add(Op.InvokeComponentLayout, (vm, { op1: _state }) => {
 });
 
 APPEND_OPCODES.add(Op.DidRenderLayout, (vm, { op1: _state }) => {
-  let { manager, state, capabilities } = check(vm.fetchValue(_state), CheckComponentInstance);
+  let instance = check(vm.fetchValue(_state), CheckComponentInstance);
+  let { manager, state, capabilities } = instance;
   let bounds = vm.elements().popBlock();
 
-  let mgr = check(manager, CheckInterface({ didRenderLayout: CheckFunction }));
+  if (vm.env.debugRenderTree !== undefined) {
+    if (hasCustomDebugRenderTreeLifecycle(manager)) {
+      let nodes = manager.getDebugCustomRenderTree(instance.definition.state, state, EMPTY_ARGS);
 
-  mgr.didRenderLayout(state, bounds);
+      nodes.reverse().forEach((node) => {
```

Comment:
Why in reverse? Also, do we really want to mutate the array directly?

Reply:
It's in reverse because we want to pop off arrays from the opposite order that they were pushed on, and reversing is fine as we return a new array every time